pages/project/[projectIdSlug]/index.tsx

```diff
@@ -1,11 +1,16 @@
 import { client } from '@/apollo/apolloClient';
 import { FETCH_PROJECT_BY_SLUG } from '@/apollo/gql/gqlProjects';
 import { IProject } from '@/apollo/types/types';
+import NotAvailableProject from '@/components/views/project/NotAvailableProject';
 
 import ProjectIndex from '@/components/views/project/ProjectIndex';
 
 const ProjectRoute = (props: { project?: IProject }) => {
-	return <ProjectIndex project={props.project} />;
+	return props.project ? (
```

Comment:
Have you considered `draft` projects? this way no one can view his/her draft projects anymore.